File: Java/SieveOfEratosthenes.java
import java.util.Scanner;
import java.util.ArrayList;

public class Main {
    public static ArrayList<Integer> SieveOfEratosthenes(int lower, int upper) {
        boolean[] isPrime = new boolean[upper+1];
        for (int i = 2; i<=upper; i++) {
            isPrime[i] = true;
        }
        isPrime[0] = false; isPrime[1] = false;
        ArrayList<Integer> primes = new ArrayList<Integer>();
        for (int i = 2; i<=upper; i++) {
            if (isPrime[i]) {
                for (int j = i+i; j<=upper; j+=i) {
                    isPrime[j] = false;
                } 
               if (i >= lower) primes.add(i);
            }
        }
        return primes;
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.println("This method finds all prime numbers between a range");
        System.out.println("Enter the lower bound of the range");
        int lower = input.nextInt();
        System.out.println("Enter the lower bound of the range");
        int upper = input.nextInt();
        ArrayList<Integer> primes = SieveOfEratosthenes(lower, upper);
        System.out.println("A list of all of the prime numbers between the given range:");
        System.out.print(primes.get(0));
        for (int i = 1; i<primes.size(); i++) {
            System.out.print(", "+primes.get(i));
        }
    }
}

File: Python/SieveOfEratosthenes.py
def sieve_eratosthenes(lower, upper):
    primes = []
    is_prime = []
    for i in range(0, upper+1):
        is_prime.append(True)
    is_prime[0], is_prime[1] = False, False

    for i in range(2, upper+1):
        if is_prime[i]:
            for j in range(i*2, upper+1, i):
                is_prime[j] = False
            if i >= lower:
                primes.append(i)
    return primes


print("This method finds all prime numbers between a range")
lower = int(input("Enter the lower bound of the range: "))
upper = int(input("Enter the upper bound of the range: "))
primes = sieve_eratosthenes(lower, upper)
print("A list of all of the primes is the following:")
print(", ".join(map(str, primes)))
